From 9ae73a6273461361eef7e83d48aadee111d6616e Mon Sep 17 00:00:00 2001
From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Sun, 23 Dec 2018 03:06:47 +0900
Subject: [PATCH 10/34] sd-device: fix segfault when error occurs in
 device_new_from_{nulstr,strv}()

As devpath may not be set yet.

When debug logging is enabled, log_device_*() calls
sd_device_get_sysname(). So, we should not assume that devpath is always
set.

Fixes #11258.

(cherry picked from commit 18fee12a2d489378a2a9b647db0d0eb8c43f5362)
---
 src/libsystemd/sd-device/sd-device.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/libsystemd/sd-device/sd-device.c b/src/libsystemd/sd-device/sd-device.c
index db58615df5..9b1ef44480 100644
--- a/src/libsystemd/sd-device/sd-device.c
+++ b/src/libsystemd/sd-device/sd-device.c
@@ -1002,6 +1002,9 @@ static int device_set_sysname(sd_device *device) {
         const char *pos;
         size_t len = 0;
 
+        if (!device->devpath)
+                return -EINVAL;
+
         pos = strrchr(device->devpath, '/');
         if (!pos)
                 return -EINVAL;
-- 
2.20.1

